Create post (#5163)
sourcegraph/about: blogposts/2022/code-insights-ga-announcement.md

---
title: 'Announcing Code Insights: analytics for engineering teams to understand and visualize their codebase over time'
author: Quinn Slack
description: 'Devs, engineering managers, and leadership across all companies and industries are looking for a better understanding of their codebase. Code Insights allows you to create insights for anything you can search for in seconds.'

> publishDate: 2022-02-24T12:00-00:00

Author: ```suggestion
publishDate: 2022-03-10T12:00-00:00
```

We started exploring Code Insights because our engineering teams faced the same problems. And it was an exciting extension of the Code Search functionality we’d already built: What if you could take anything you could search for and see how the result has changed over time? We know the truth is in the code, so the idea of unleashing this information for engineering teams and their leaders seemed very compelling.

Then, as we built from concept to prototype to Beta to General Availability today, we were consistently surprised by how eager our customers were to start using it. Some of them would even snoop around our GitHub and changelog drafts to find out what features would arrive in each release before we shared the updates. It turns out we weren’t the only ones who wanted a tool to finally set KPIs on custom metrics based directly on the source of truth—the code—rather than manual spreadsheets or ad hoc updates.

<figure>
  <img src="https://storage.googleapis.com/sourcegraph-assets/blog/code-insights-ga-blogs/code-insights-dashboard.svg" alt="Code Insights dashboard" class="no-shadow">
</figure>

## Beyond code search

> Code Insights has huge momentum after an incredible year of growth at Sourcegraph. More than 1.275 million devs used Sourcegraph over the last 12 months, including many of the world's leading engineering teams, like 3/5 FAANG, Reddit, Uber, GE, PayPal, and Dropbox. Those teams first came to Sourcegraph for Code Search, which remains foundational to everything we do as we’ve expanded our product with Code Insights and [Batch Changes](https://about.sourcegraph.com/blog/introducing-batch-changes/).

Author: @amieroth Noting that this line in particular needs review/revision before the launch

Reviewer: @rebeccadee thoughts on changing the first sentence to "Code Insights is a core part of our growth this year. More than 1.275. million..." I think the rest works fine, but LMK if there's anything in particular you feel is off given the more neutral tone we're looking for.

Author: ```suggestion
Code Insights is a core part of our growth this year. More than 1.275 million devs used Sourcegraph over the last 12 months, including many of the world's leading engineering teams, like 3/5 FAANG, Reddit, Uber, GE, PayPal, and Dropbox. Those teams first came to Sourcegraph for Code Search, which remains foundational to everything we do as we’ve expanded our product with Code Insights and [Batch Changes](https://about.sourcegraph.com/blog/introducing-batch-changes/).
```
